aggsender/validator/validator_service.go

```diff
@@ -2,22 +2,131 @@ package validator
 
 import (
 	"context"
+	"fmt"
 
+	v1types "buf.build/gen/go/agglayer/interop/protocolbuffers/go/agglayer/interop/types/v1"
+	"github.com/agglayer/aggkit"
+	"github.com/agglayer/aggkit/agglayer"
+	agglayergrpc "github.com/agglayer/aggkit/agglayer/grpc"
+	agglayertypes "github.com/agglayer/aggkit/agglayer/types"
+	"github.com/agglayer/aggkit/aggsender/types"
 	v1 "github.com/agglayer/aggkit/aggsender/validator/proto/v1"
-	"github.com/agglayer/aggkit/log"
+	aggkitcommon "github.com/agglayer/aggkit/common"
+	"github.com/agglayer/aggkit/grpc"
+	signertypes "github.com/agglayer/go_signer/signer/types"
+	"github.com/ethereum/go-ethereum/common"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/types/known/emptypb"
 )
 
 // ValidatorService implements the gRPC server for the AggsenderValidator service.
 type ValidatorService struct {
 	// Embed the generated server interface to ensure forward compatibility
 	v1.UnimplementedAggsenderValidatorServer
+
+	validator      types.CertificateValidator
+	agglayerClient agglayer.AggLayerClientCertificateIDQuerier
+	signer         signertypes.Signer
+	log            aggkitcommon.Logger
+}
+
+func NewValidatorService(
+	logger aggkitcommon.Logger,
+	validator types.CertificateValidator,
+	agglayerClient agglayer.AggLayerClientCertificateIDQuerier,
+	signer signertypes.Signer) *ValidatorService {
+	return &ValidatorService{
+		log:            logger,
+		validator:      validator,
+		agglayerClient: agglayerClient,
+		signer:         signer,
+	}
+}
+
+// Implementa el método Status
```

Comment:
Although I understand, maybe translate it to English (or remove the comment) :slightly_smiling_face:
```suggestion
// Implementa el método Status
```

Reply:
you're right!!! 🤣🤣